website/ruchy.php

```diff
@@ -89,13 +89,13 @@
 // autopoprawione...
 $g_lon = $_GET['lon'];
 // autopoprawione...
-$g_nr = $_GET['nr'];
+$g_nr = preg_replace('/[^a-z0-9]+/i', '', $_GET['nr']);
```

Comment:
There was already a check below, wasn't it?
```
        if (!ctype_alnum($kret_nr)) {
```

I mean, doing replace feels like a wrong way to workaround such potential problems. If it's a number (like `lon`) then it makes sense to cast it to number (which was not done as I can see), but for whatever string it looks odd. Also it will lead to different handling for string-names and string-ids (at least because you don't want to forbid people to use `<` in GK names). Isn't it probably better to just do proper escaping while outputting/making sql queries and don't worry about pre-processing?

Reply:
The check you mention bellow will raise a form validation error, but the provided content will still be re-displayed as is to the user, which then may produce XSS. https://www.openbugbounty.org/reports/1398772/

The current website is very messy, I don;t want to spend lot of effort on this branch, and wish to concentrate on the future (branch `new-theme`). I think this quick and simple fix will throw away the reported problem without breaking the feature.

It was deployed on https://staging.geokrety.org, you can cross check if all is good :) I was about to deploy it in production, but I'll wait a bit from your input. @okainov Would you propose another approach via PR?

Reply:
>The current website is very messy, I don;t want to spend lot of effort on this branch, and wish to concentrate on the future (branch new-theme). I think this quick and simple fix will throw away the reported problem without breaking the feature.

I mean, if it's considered as quick fix for legacy code that is not really maintained any more with the migration to the new codebase - it's fine, no objections. Just to make it clear that it's clearly legacy-intended fix then :)

Reply:
> it's considered as quick fix for legacy code that is not really maintained any more

I confirm :+1: then let's deploy to prod.